website/docs/developer/research/research-orca-delete-functionality.md

```diff
@@ -0,0 +1,122 @@
+Check API gateway [examples](https://github.com/nasa/cumulus-orca/tree/develop/modules/api-gateway) currently used by ORCA. 
+
+### Lambda function
+
+The delete functionality can be performed by two lambda functions based on initial research. One of the lambda functions will be used to perform soft delete- delete from S3 bucket but only flag the granule in the database for deletion after 90 days. The lambda should also verify with CMR and/or Cumulus that the granule has been deleted from those locations prior to allowing the delete to proceed in ORCA. The lambda will be triggered by an API call made by the user once the user is authenticated with Launchpad.
+
+The other lambda function can be used to perform a physical delete- delete from S3 glacier as well as from the database. Similar to soft delete, it should first verify with CMR and/or Cumulus that the granule has been deleted from those locations before deleting from ORCA. This will require further discussion with Cumulus team before performing the actual work.
+
+
+### Deleting objects from AWS S3
+
+To delete an object from an S3 bucket, the [delete_objects](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.delete_objects)
+can be used from the boto3 library. An example of the usage can be seen below.
+Some notable items if this library is used includes the following.
+- It can be used to delete up to 1000 objects from a bucket using a single API call.
+- When performing this action on an MFA Delete enabled bucket that attempts to delete any versioned objects, an MFA token must be included orelse it will fail.
+- If the object specified in the request is not found, Amazon S3 returns the result as deleted.
+- Each DeleteObjects API incurs a little cost.
+
+#### Example Using delete_objects
+
+```python
+import boto3
+
+s3_client = boto3.client('s3')
+bucket = "riz-orca"
+response = s3_client.delete_objects(
+    Bucket=bucket,
+    Delete={
+        'Objects': [
+            {
+                'Key': 'MOD09A1.006/MOD09A1.A2019025.h07v03.006.2019037205734.hdf',
+            }
+        ]
+    }
+)
+print(response)
+
+# Results
+{
+   "ResponseMetadata":{
+      "RequestId":"KJCZAPM0RWMJNNC7",
+      "HostId":"z2fRlgMxT10ApzcyDOLp3REn3PjM71OSgdH0oPpe4v2pHchmSTPCfEjBjTx00PyTL55wgmQ7gF4=",
+      "HTTPStatusCode":200,
+      "HTTPHeaders":{
+         "x-amz-id-2":"z2fRlgMxT10ApzcyDOLp3REn3PjM71OSgdH0oPpe4v2pHchmSTPCfEjBjTx00PyTL55wgmQ7gF4=",
+         "x-amz-request-id":"KJCZAPM0RWMJNNC7",
+         "date":"Tue, 08 Mar 2022 05:19:07 GMT",
+         "content-type":"application/xml",
+         "transfer-encoding":"chunked",
+         "server":"AmazonS3",
+         "connection":"close"
+      },
+      "RetryAttempts":0
+   },
+   "Deleted":[
+      {
+         "Key":"MOD09A1.006/MOD09A1.A2019025.h07v03.006.2019037205734.hdf"
+      }
+   ]
+}
+```
+### Logging
+
```

Comment:
What we should probably talk about here more is auditing. Would this be something we log or store in the database as an audit trail? How long should we or do we need to keep those audit records? How long should we keep the logs? What makes sense or what is current policies?

Reply:
I assume we store this in s3 for about a month but the longer it is stored, the more we get charged for storage. Seems like a question for the bigger team before coming to a conclusion.

Reply:
Make sure to add that to your questions list and/or assumptions in the document.

Reply:
added